packages/flutter/lib/src/cupertino/search_field.dart

```diff
@@ -0,0 +1,326 @@
+  final Color? itemColor;
+
+  /// Sets the base icon size for the suffix and prefix icons.
+  ///
+  /// The size of the icon is scaled using the accessibility font
+  /// scale settings. Defaults to [20.0].
+  final double itemSize;
+
+  /// Sets the padding insets for the suffix.
+  ///
+  /// Defaults to padding that replicates the `UISearchTextField` suffix
+  /// look. The inset values were determined using the comparison tool in
+  /// https://github.com/flutter/platform_tests/.
+  final EdgeInsets prefixInsets;
+
+  /// Sets the padding insets for the prefix.
+  ///
+  /// Defaults to padding that replicates the `UISearchTextField` prefix
+  /// look. The inset values were determined using the comparison tool in
+  /// https://github.com/flutter/platform_tests/.
+  final EdgeInsets suffixInsets;
+
+  /// Sets the suffix widget's icon.
+  ///
+  /// Defaults to the X-Mark [CupertinoIcons.xmark_circle_fill]. The suffix is
+  /// customizable so that users can override it with other options, like a
+  /// bookmark icon.
+  final Icon suffixIcon;
+
+  /// Dictates when the X-Mark (suffix) should be visible.
+  ///
+  /// Defaults to only on when editing.
+  final OverlayVisibilityMode suffixMode;
+
+  /// Sets the X-Mark (suffix) action.
+  ///
+  /// Defaults to clearing the text. The suffix action is customizable
+  /// so that users can override it with other functionality, that isn't
+  /// necessarily clearing text.
+  final VoidCallback? onSuffixTap;
+
+  @override
+  Widget build(BuildContext context) {
+    // The icon size will be scaled by a factor of the accessibility text scale,
+    // to follow the behavior of `UISearchTextField`.
+    final double scaledIconSize =
+        MediaQuery.textScaleFactorOf(context) * itemSize;
+
+    // If decoration was not provided, create a decoration with the provided
+    // background color and border radius.
+    final BoxDecoration decoration = this.decoration ??
+        BoxDecoration(
+          color: CupertinoDynamicColor.resolve(
+              backgroundColor ?? CupertinoColors.tertiarySystemFill, context),
+          borderRadius:
+              borderRadius ?? const BorderRadius.all(Radius.circular(9.0)),
+        );
+
+    final TextStyle placeholderStyle = this.placeholderStyle ??
+        TextStyle(
+          color: CupertinoDynamicColor.resolve(
+              CupertinoColors.secondaryLabel, context),
+        );
+
+    final IconThemeData iconThemeData = IconThemeData(
+        color: CupertinoDynamicColor.resolve(itemColor, context),
+        size: scaledIconSize);
+
+    final Widget prefix = Padding(
+      child: IconTheme(
+          child: const Icon(CupertinoIcons.search), data: iconThemeData),
+      padding: prefixInsets,
+    );
+
+    // TODO(DanielEdrisian): Replace [GestureDetector] with a [CupertinoButton].
+    // The reason why I went with [GestureDetector] was because
+    // [CupertinoButton] was messing with the content size of the entire
```

Comment:
Does setting its padding and minSize to 0 work?

Reply:
minSize did it!